awx/main/scheduler/task_manager.py

```diff
@@ -283,11 +283,13 @@ def create_project_update(self, task, project_id=None):
         logger.debug('Spawned {} as dependency of {}'.format(project_task.log_format, task.log_format))
         return project_task
 
-    def create_inventory_update(self, task, inventory_source_task):
+    def create_inventory_update(self, task, inventory_source_task, should_set_scm_branch=False):
         inventory_task = InventorySource.objects.get(id=inventory_source_task.id).create_inventory_update(_eager_fields=dict(launch_type='dependency'))
 
         inventory_task.created = task.created - timedelta(seconds=2)
         inventory_task.status = 'pending'
+        if should_set_scm_branch:
+            inventory_task.scm_branch = task.scm_branch
```

Comment:
Even if it's more verbose, I have a gut feeling that this method should evaluate whether or not to set scm_branch from what's already available in `task` and `inventory_source_task`.

BTW, the naming of `inventory_source_task` bothers me, right? It seems to be an InventorySource, so maybe we could just call it `inventory_source`?